generators/server/templates/src/main/resources/config/_application.yml

```diff
@@ -123,7 +123,7 @@ jhipster:
         corePoolSize: 2
         maxPoolSize: 50
         queueCapacity: 10000
-    #cors: #By default CORS are not enabled. Uncomment to enable.
+    #cors: #By default CORS is disabled. Uncomment to enable.
```

Comment:
OK, so we have a `CorsConfiguration` bean created anyway, I didn't know this -> in that case you can inject it in the constructor (it's mandatory), and only add it if it's configured? That was basically your original code, only difference is that it's created all the time